server/src/main/java/org/opensearch/snapshots/RestoreService.java

```diff
@@ -713,7 +774,9 @@ private void validateExistingIndex(
                     private IndexMetadata updateIndexSettings(
                         IndexMetadata indexMetadata,
                         Settings changeSettings,
-                        String[] ignoreSettings
+                        String[] ignoreSettings,
+                        Settings changeSettingsInternal,
+                        String[] ignoreSettingsInternal
```

Comment:
any reason why we created new arguments here? can we merge with changeSettings and ignoreSettings ?